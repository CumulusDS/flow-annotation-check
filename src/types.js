/**
 * @flow
 */

export type Args = {
  absolute?: boolean,
  exclude?: Array<string>,
  include?: Array<string>,
  root?: string,
};

export type Flags = {
  absolute: boolean,
  exclude: Array<string>,
  include: Array<string>,
  root: string,
};

export type FlowStatus = 'flow' | 'flow weak' | 'no flow';

export type StatusEntry = {
  file: string,
  status: FlowStatus,
};

export type ValidityEntry = {
  file: string,
  isValid: boolean,
  status: FlowStatus,
  threwError: boolean,
};

export type StatusReport = Array<StatusEntry>;
export type ValidationReport = Array<ValidityEntry>;
export type ErrorReport = Array<string>;
