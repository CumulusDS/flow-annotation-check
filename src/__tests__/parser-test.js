'use strict';

/**
 * @flow
 */

import path from 'path';
import getParser from '../parser';
import {DEFAULT_FLAGS} from '../types';

describe('getParser', () => {
  it('should print the help message', () => {
    expect(getParser().formatHelp()).toMatchSnapshot();
  });
});
