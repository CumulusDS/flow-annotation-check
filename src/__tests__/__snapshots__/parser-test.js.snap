// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getParser should print the help message 1`] = `
"usage: PROG [-h] [-v] [-f FLOW_PATH]
            [-o {text,html-table,csv,junit,summary,json}] [--show-summary]
            [--summary-only]
            [--list-files {all,flow,flowstrict,flowstrictlocal,noflow,flowweak,none}]
            [--html-file HTML_FILE] [--csv-file CSV_FILE]
            [--junit-file JUNIT_FILE] [--json-file JSON_FILE]
            [--summary-file SUMMARY_FILE] [-a]
            [--level {any,flowweak,flow,flowstrictlocal,flowstrict}]
            [--allow-weak] [--require-strict-local] [--require-strict]
            [-i INCLUDE] [-x EXCLUDE] [--validate]
            [root]

positional arguments:
  root                  The root directory to glob files from. (default:
                        '\\".\\"')

optional arguments:
  -h, --help            show this help message and exit
  -v, --version         show program's version number and exit
  -f FLOW_PATH, --flow-path FLOW_PATH
                        The path to the flow command. (default: '\\"flow\\"')
  -o {text,html-table,csv,junit,summary,json}, --output {text,html-table,csv,junit,summary,json}
                        Output format for status/filename pairs. (default:
                        '\\"text\\"')
  --show-summary        Include summary data. Does not apply to saved file
                        output or jUnit output. (default: 'false')
  --summary-only        Unused. Switch to --show-summary instead.
  --list-files {all,flow,flowstrict,flowstrictlocal,noflow,flowweak,none}
                        Filter the list of files based on the reported status.
                        See '--level=flowweak' or '--allow-weak' to control
                        when flow-weak files are included or excluded from the
                        'flow' or 'noflow' checks. (default: '\\"all\\"')
  --html-file HTML_FILE
                        Save the html table output directly into HTML_FILE.
                        (default: 'null')
  --csv-file CSV_FILE   Save CSV output directly into CSV_FILE. (default:
                        'null')
  --junit-file JUNIT_FILE
                        Save jUnit output directly into JUNIT_FILE. (default:
                        'null')
  --json-file JSON_FILE
                        Save JSON output directly into JSON_FILE. (default:
                        'null')
  --summary-file SUMMARY_FILE
                        Save a text-format summary of the report into
                        SUMMARY_FILE. (default: 'null')
  -a, --absolute        Report absolute path names. (default: 'false')
  --level {any,flowweak,flow,flowstrictlocal,flowstrict}
                        The minimum strictness required in each file. Levels
                        progress in ascending order through: 'any > weak >
                        flow > strict-local > strict'. The program will exit 1
                        if any file has a level lower than this setting.
                        (default: '\\"flow\\"')
  --allow-weak          Alias for '--level=flowweak'. Consider '@flow weak' as
                        an accepable annotation. See
                        https://flowtype.org/docs/existing.html#weak-mode for
                        reasons why this should only be used temporarily.
                        (default: 'false')
  --require-strict-local
                        Alias for '--level=flowstrictlocal'. Consider '@flow
                        strict-local' as the minimum allowable level.
                        (default: 'false')
  --require-strict      Alias for '--level=flowstrict'. Consider '@flow
                        strict' as the only accepable annotation. (default:
                        'false')
  -i INCLUDE, --include INCLUDE
                        Glob for files to include. Can be set multiple times.
                        (default: '[\\"**/*.js\\"]')
  -x EXCLUDE, --exclude EXCLUDE
                        Glob for files to exclude. Can be set multiple times.
                        (default:
                        '[\\"+(node_modules|build|flow-typed)/**/*.js\\"]')
  --validate            Run in validation mode. This injects errors into
                        globbed files and checks the flow-annotation status.
"
`;
